src/tensors/cpu/wasm_intgemm_interface.h

```diff
@@ -0,0 +1,193 @@
+#pragma once
+
+/** Main interface for integer matrix multiplication followed by addition of bias for wasm.
+ *
+ * C = A * B + Bias
+ *
+ * A is typically activations whose rows should be a multiple of 1 (i.e. no restriction) and
+ * columns should be a multiple of 64.
+ *
+ * B is typically fixed model parameters whose rows should be a multiple of 64 and columns
+ * should be a multiple of 8.
+ *
+ * All matrices A, B and C are in row-major format.
+ *
+ * Please note that most of the functions in this interface might have architecture specific
+ * implementations.
+ */
+
+#include <cstdint>
+
+using Index = uint32_t;
+
+/**
+ * Prepare B for the Matrix Multiply routine.
+ *
+ * B is prepared in a CPU-dependent format by performing quantization on floating values.
+ * Please note that this interface might have architecture specific implementations.
+ *
+ * @param[in]   input_B                An array representing the input 2-D matrix.
+ *                                     Size of the array = `output_rows` * `output_cols`.
+ *
+ *                                     If the input matrix is in transposed form then:
+ *                                     Shape of the matrix: (`output_cols`, `output_rows`)
+ *
+ *                                     If the input matrix is NOT in transposed form then:
+ *                                     Shape of the matrix: (`output_rows`, `output_cols`)
+ * @param[in]   scale                  The scaling factor (for quantization)
+ * @param[in]   zero_point             The zero point (for quantization)
+ * @param[in]   is_input_transposed    Whether the input matrix is in transposed form or not.
+ * @param[in]   output_rows            No. of rows of output (prepared B) matrix.
+ *                                     It should be a multiple of 64.
+ * @param[in]   output_cols            No. of columns of output (prepared B) matrix.
+ *                                     It should be a multiple of 8.
+ * @param[out]  output                 An array representing the prepared B matrix in row-major
+ *                                     format. Size of the array = `output_rows` * `output_cols`.
+ *                                     Shape of the matrix: (`output_rows`, `output_cols`)
+ */
+void int8PrepareB(const float* input_B,
+                  float scale,
+                  int8_t zero_point,
+                  bool is_input_transposed,
+                  Index output_rows,
```

Comment:
If it applies to both the output and input, why is it called output?

Reply:
Also why is the user specifying the shape of an opaque output representation that is neither row nor column major?

Reply:
@abhi-agg we should have the comment, pointing out that if there is a transposition happening, the rows/cols is reversed. This is typical in BLAS style interfaces, so users will be familiar with this peculiarity. Refer to this naming:

https://github.com/kpu/intgemm/blob/master/intgemm/intgemm.h#L339

Reply:
@XapaJIaMnu Please check out the latest version. I have tried to be descriptive there. If that still doesn't feel enough, please feel free to comment and I will address it 👍

Reply:
Resolving it for now.